### PR TITLE
Create C100 address common component

### PR DESCRIPTION
**Pull request by hcarslaw**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] the enable_keep_helm label has been added, if the helm release should be persisted after a successful build

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/steps/c100-rebuild/address/manual/ManualAddressPostController.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+    if (onlycontinue) {
+      req.session.errors = form.getErrors(formData);
+      return this.redirect(req, res);
+    } else if (saveAndComeLater) {
+      this.saveAndComeLater(req, res, req.session.userCase);
+    }
+  }
+
+  private updateAddressDetails = (req: AppRequest<AnyObject>, id: string, partyType: C100UrlPartyType, formData) => {
```
Comment (vivek-sekhar) [inline src/main/steps/c100-rebuild/address/manual/ManualAddressPostController.ts:53]: return type missing

Comment (vivek-sekhar) [reply src/main/steps/c100-rebuild/address/manual/ManualAddressPostController.ts:53]: i don't think you need to return anything

File: src/main/steps/c100-rebuild/address/select/SelectAddressPostController.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+      getUpdatedForm(req.session.userCase, id, partyType as C100UrlPartyType, selectAddressForm as C100AddressForm)
+        .fields as FormFields
+    );
+    const { onlycontinue, ...formFields } = req.body;
+    const { _csrf, ...formData } = form.getParsedBody(formFields);
+    const selectedAddressIndex = Number(formData['selectAddress']);
+
+    if (selectedAddressIndex >= 0) {
+      const selectedAddress = req.session.addresses[selectedAddressIndex];
+      req = this.updateAddressDetails(req, id, partyType as C100UrlPartyType, selectedAddress, selectedAddressIndex);
```
Comment (vivek-sekhar) [inline src/main/steps/c100-rebuild/address/select/SelectAddressPostController.ts:32]: same here

**Review by vivek-sekhar — CHANGES_REQUESTED** [review]

File: src/main/steps/c100-rebuild/address/manual/ManualAddressPostController.ts (hunk 1 above)
Comment (hcarslaw) [reply src/main/steps/c100-rebuild/address/manual/ManualAddressPostController.ts:53]: removed the return statement in updateAddressDetails

File: src/main/steps/c100-rebuild/address/select/SelectAddressPostController.ts (hunk 2 above)
Comment (hcarslaw) [reply src/main/steps/c100-rebuild/address/select/SelectAddressPostController.ts]: removed the return statement in updateAddressDetails

**Review by vivek-sekhar — APPROVED** [review]

**Pull request merged**